txpipe/covariance.py

```diff
@@ -788,12 +813,22 @@ def get_workspaces_dict(self, cl_sacc, masks_names):
                 hashes[m] = cl_sacc.metadata[f"hash/{m}"]
         ell_hash = cl_sacc.metadata["hash/ell_hash"]
 
-        w = {}
+        w = {'00': {}, '02': {}, '22': {}}
+        # Get the number of data points per Cell
+        dtype = cl_sacc.get_data_types()[0]
+        trs = cl_sacc.get_tracer_combinations()[0]
+        ell_eff, _ = cl_sacc.get_ell_cl(dtype, *trs)
+        n_ell = ell_eff.size
         for tr1, tr2 in cl_sacc.get_tracer_combinations():
+            # This assumes that the name of the tracers will be 'lens'or
```

Comment:
I think it actually assumes that the name will include "lens" or "source", not actually *be* that.